posture/serialnumber_notmacos.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+//go:build !darwin
+
+package posture
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/digitalocean/go-smbios/smbios"
+	"tailscale.com/util/multierr"
+)
+
+// getByteFromSmbiosStructure retrieves a 8-bit unsigned integer at the given offset.
+func getByteFromSmbiosStructure(s *smbios.Structure, offset int) uint8 {
+	// the `Formatted` byte slice is missing the first 4 bytes of the structure that are stripped out as header info.
+	// so we need to subtract 4 from the offset mentioned in the SMBIOS documentation to get the right value.
+	index := offset - 4
+	if index >= len(s.Formatted) {
+		return 0
+	}
+
+	return s.Formatted[index]
+}
+
+// getStringFromSmbiosStructure retrieves a string at the given offset.
+// Returns an empty string if no string was present.
+func getStringFromSmbiosStructure(s *smbios.Structure, offset int) (string, error) {
+	index := getByteFromSmbiosStructure(s, offset)
+
+	if index == 0 || int(index) > len(s.Strings) {
+		return "", errors.New("offset does not exist in smbios structure")
+	}
+
+	str := s.Strings[index-1]
+	trimmed := strings.TrimSpace(str)
+
+	// Convert to lowercase to address multiple formats:
+	//   - "To Be Filled By O.E.M."
+	//   - "To be filled by O.E.M."
+	if strings.ToLower(trimmed) == "to be filled by o.e.m." {
```

Comment:
This could be a prefix check instead, "to be filled", maybe something else is in the wild out there.